taxonomy/tests/test_api.py

```diff
@@ -0,0 +1,346 @@
+    get_term,
+    get_terms_for_learning_resource,
+    get_types_for_vocabulary,
+    get_vocabularies_for_type,
+    get_vocabulary,
+    remove_term_from_learning_resource,
+    remove_type_from_vocabulary,
+    TaxonomyAPIException,
+)
+
+
+class TestApi(TestCase):
+    """Tests for taxomony API"""
+
+    def setUp(self):
+        super(TestApi, self).setUp()
+        self.user = User.objects.create(username="test")
+
+        self.repository = Repository.objects.create(
+            create_date="2014-08-08",
+            created_by=self.user,
+        )
+
+        self.vocabulary = Vocabulary.objects.create(
+            repository=self.repository,
+            required=False,
+            weight=100,
+        )
+
+        self.learning_resource_type1 = LearningResourceType.objects.create(
+            name="Type1",
+        )
+        self.learning_resource_type2 = LearningResourceType.objects.create(
+            name="Type2",
+        )
+
+        self.course = Course.objects.create(
+            repository=self.repository,
+            imported_by=self.user,
+        )
+
+        self.learning_resource = LearningResource.objects.create(
+            course=self.course,
+            learning_resource_type=self.learning_resource_type1,
+        )
+
+        self.term = Term.objects.create(
+            vocabulary=self.vocabulary,
+            weight=4,
+        )
+
+    def test_add_term_to_learning_resource(self):
+        """
+        Test add_term_to_learning_resource
+        """
+        self.assertEquals(0, self.learning_resource.terms.count())
+        add_term_to_learning_resource(self.learning_resource.id, self.term.id)
+        self.assertEquals([self.term],
+                          list(self.learning_resource.terms.all()))
+        add_term_to_learning_resource(self.learning_resource.id, self.term.id)
+
+        # duplicate term is ignored
+        self.assertEquals([self.term],
+                          list(self.learning_resource.terms.all()))
+
+        self.assertRaises(LearningResource.DoesNotExist,
+                          lambda: add_term_to_learning_resource(
+                              -3, self.term.id))
+        self.assertRaises(Term.DoesNotExist,
+                          lambda: add_term_to_learning_resource(
+                              self.learning_resource.id, -4))
+        self.assertRaises(LearningResource.DoesNotExist,
+                          lambda: add_term_to_learning_resource(
+                              None, None))
+
+    def test_add_type_for_vocabulary(self):
+        """
+        Test add_type_for_vocabulary
+        """
+        self.assertEquals(0, self.vocabulary.learning_resource_types.count())
+        add_type_for_vocabulary(self.learning_resource_type1.name,
+                                self.vocabulary.id)
+        self.assertEquals([self.learning_resource_type1],
+                          list(self.vocabulary.learning_resource_types.all()))
```

Comment:
I like these tests. Normally I would check that the `count()` had increased by the exact number I expected, but this kills three birds with one stone by asserting the count is correct, what we think is in the database is what's actually there, and that there's nothing unexpected in the database.